packages/application-extension/src/index.tsx

```diff
@@ -227,11 +235,11 @@ const router: JupyterLabPlugin<IRouter> = {
 /**
  * The tree route handler provider.
  */
-const tree: JupyterLabPlugin<void> = {
+const tree: JupyterFrontEndPlugin<void> = {
   id: '@jupyterlab/application-extension:tree',
   autoStart: true,
-  requires: [IRouter],
-  activate: (app: JupyterLab, router: IRouter) => {
+  requires: [JupyterLab.IInfo, IRouter],
```

Comment:
If this requires a token from the `JupyterLab` namespace, should it also check if the app is an instance of `JupyterLab`? Or do you see this being useful with other potential frontends? If that's the case, is there a useful subset of `IInfo` that can be made non-lab-specific?

Reply:
I think we should make the `urls` and possibly `directories` segments have their own tokens because they're probably useful for others. I didn't know if it needs to be done here, though. What do you think?